server/src/main/java/org/elasticsearch/common/geo/GeoJson.java

```diff
@@ -610,4 +617,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         }
     }
 
+    public static Map<?, ?> toXContentMap(Geometry geometry) throws IOException {
```

Comment:
Probably worth javadoc. I was confused by the method name because I usually think of `map` and `xcontent` as mutually exclusive representations.

Reply:
also, XContent here is just an implementation detail. it is a normal Map, right?

Reply:
I will clean this up. It seems I had a different notion of what exactly `xContent` means.